lib/r7b-impl/base.sls

```diff
@@ -291,17 +300,79 @@ write-u8 zero?
     (itr 0)
     ret))
 
-(define (vector-copy! . args)
-  (raise "vector-copy! not supported"))
+;; vector-copy! bytevector-copy! bytevector-append from　https://github.com/okuoku/yuni.
+(define (vector-copy!/itr+ to at from start end)
+  (unless (= start end)
+    (vector-set! to at (vector-ref from start))
+    (vector-copy!/itr+ to (+ 1 at) from (+ 1 start) end)))
 
-(define (bytevector-append . args)
-  (raise "bytevector-append not supported"))
+(define (vector-copy!/itr- to at from start end)
+  (vector-set! to at (vector-ref from end))
+  (unless (= start end)
+    (vector-copy!/itr- to (- at 1) from start (- end 1))))
 
-(define (bytevector . args)
-  (raise "bytevector not supported"))
+(define vector-copy!
+  (case-lambda
+    ((to at from)
+     (vector-copy! to at from 0
+                   (min (- (vector-length to) at) (vector-length from))))
+    ((to at from start)
+     (vector-copy! to at from start
+                   (min (+ start (- (vector-length to) at))
+                        (vector-length from))))
+    ((to at from start end)
+     (unless (= start end)
+       (cond
+         ((and (eq? from to)
+               (< start at end))
+          (vector-copy!/itr- to (+ (- end start) at) from start (- end 1)))
+         (else
+           (vector-copy!/itr+ to at from start end)))))))
+
+(define (bytevector-append . bvs)
+  (u8-list->bytevector (apply append (map bytevector->u8-list bvs))))
+
+(define (bytevector . elm*)
+  (u8-list->bytevector elm*))
```

Comment:
These aren't so efficient as well; if we had any O(N) on Scheme side, it'd be better to be in implement it in native.

Reply:
Agreed. I'll keep them as is for now.